src/realm/query_engine.hpp

```diff
@@ -425,6 +425,17 @@ class IntegerNode : public IntegerNodeBase<LeafType> {
 
     size_t find_first_local(size_t start, size_t end) override
     {
+        TConditionFunction c;
```

Comment:
All our operators shouldn't have anything going on in the constructor, but can you move this to inside the scope where it is used so that one isn't constructed/destructed when it isn't used? (Hopefully the compiler knows that this is a no-op but you never know)